### PR TITLE
[Telink] Implement OnSubscriptionRequested callback for SED examples

### PR DESCRIPTION
**Pull request by s07641069**
**Change overview**

Copied implementation from https://github.com/project-chip/connectedhomeip/pull/26130.
Added OnSubscriptionRequested callback implementation to make the Matter accessory device negotiate subscription report interval instead of always accepting the requested value.

**Testing**

Tested manually with chip-tool.

Steps:

- Run: $ chip-tool pairing ble-thread <...>
- Wait till success
- Run: $ chip-tool onoff on ${NODE_ID }1
- Wait till success
- Run: $ chip-tool onoff off ${NODE_ID }1
- Wait till success

### CODE REVIEW

File: examples/platform/telink/common/src/AppTaskCommon.cpp [hunk 1]
```diff
@@ -227,6 +231,10 @@ CHIP_ERROR AppTaskCommon::InitCommonParts(void)
     emberAfEndpointEnableDisable(kNetworkCommissioningEndpointSecondary, false);
 #endif
 
+#ifdef CONFIG_CHIP_ICD_SUBSCRIPTION_HANDLING
```
Comment (bzbarsky-apple) [inline examples/platform/telink/common/src/AppTaskCommon.cpp:234]: This is fine, but note that non-ICDs may also want to put a floor on their MaxInterval so they can hit their battery use targets.

**Review by bzbarsky-apple — APPROVED** [review]

File: examples/pump-controller-app/telink/CMakeLists.txt [hunk 2]
```diff
@@ -85,3 +85,7 @@ chip_configure_data_model(app
 if(CONFIG_CHIP_OTA_REQUESTOR)
     target_sources(app PRIVATE ${TELINK_COMMON}/util/src/OTAUtil.cpp)
 endif()
+
+if(CONFIG_CHIP_ICD_SUBSCRIPTION_HANDLING)
+    target_sources(app PRIVATE ${TELINK_COMMON}/util/src/ICDUtil.cpp)
```
[the remark this answers is not in the document]
Comment (s07641069) [reply examples/pump-controller-app/telink/CMakeLists.txt]: @bzbarsky-apple Good remark Thx!
Сhanges pushed, please check if I did as you mean.

File: examples/platform/telink/common/src/AppTaskCommon.cpp [hunk 3]
```diff
@@ -33,6 +33,10 @@
 #include "OTAUtil.h"
 #endif
 
+#ifdef CONFIG_CHIP_ICD_SUBSCRIPTION_HANDLING
+#include <app/InteractionModelEngine.h>
```
Comment (bzbarsky-apple) [inline examples/platform/telink/common/src/AppTaskCommon.cpp:37]: Needs to include ICDUtil.h too, right?

Comment (s07641069) [reply examples/platform/telink/common/src/AppTaskCommon.cpp:38]: @bzbarsky-apple Moved ICDUtil.h include into cpp file

**Review by bzbarsky-apple — APPROVED** [review]

File: src/platform/telink/BUILD.gn [hunk 4]
```diff
@@ -21,6 +21,7 @@ assert(chip_device_platform == "telink")
 
 static_library("telink") {
   sources = [
+    "${chip_root}/examples/platform/telink/util/src/ICDUtil.cpp",
```
Comment (Damian-Nordic) [inline src/platform/telink/BUILD.gn:24]: It is unusual to include a file from `examples` in a library defined in `src/platform`. Wouldn't it be cleaner if you moved `ICDUtil.*` to `src/platform/telink` now that it's a part of your platform library?

**Review by Damian-Nordic — APPROVED** [review]

**Comment by s07641069** [comment]
@Damian-Nordic looks like PR ready for merge.

**Pull request merged**